eventsource-client/src/event_parser.rs

```diff
@@ -281,26 +281,21 @@ impl EventParser {
     // incomplete lines from previous chunks.
     fn decode_and_buffer_lines(&mut self, chunk: Bytes) {
         let mut lines = chunk.split_inclusive(|&b| b == b'\n' || b == b'\r');
-
         // The first and last elements in this split are special. The spec requires lines to be
         // terminated. But lines may span chunks, so:
         //  * the last line, if non-empty (i.e. if chunk didn't end with a line terminator),
         //    should be buffered as an incomplete line
         //  * the first line should be appended to the incomplete line, if any
 
         if let Some(incomplete_line) = self.incomplete_line.as_mut() {
-            let line = lines
-                .next()
-                // split always returns at least one item
-                .unwrap();
-            trace!(
-                "extending line from previous chunk: {:?}+{:?}",
-                logify(incomplete_line),
-                logify(line)
-            );
+            if let Some(line) = lines.next() {
+                trace!(
+                    "extending line from previous chunk: {:?}+{:?}",
+                    logify(incomplete_line),
+                    logify(line)
+                );
 
-            self.last_char_was_cr = false;
-            if !line.is_empty() {
+                self.last_char_was_cr = false;
```

Comment:
No longer need to check `is_empty()`.

Reply:
Does the specification of the method allow for the possibility of the line being empty in the future?

Reply:
I find the wording of the method docs [somewhat confusing to parse,](https://doc.rust-lang.org/nightly/core/primitive.slice.html#method.split_inclusive) but as far as I can tell, no. 

> Returns an iterator over subslices separated by elements that match pred. The matched element is contained in the end of the previous subslice as a terminator.

If we have `Some(match)`, that match must contain the element we were splitting around, implying the length of the match is > 0. 

> If the last element of the slice is matched, that element will be considered the terminator of the preceding slice. That slice will be the last item returned by the iterator.

If we have input `\n`, and we are splitting around `\n`, then we'll receive Some(`\n`). 

If there is no match at all, the whole slice appears to be returned. I will re-read the specification again on Monday with fresh eyes.

Reply:
I added an assertion. Not sure how we feel about that. I don't want to crash customer code, but this is a guarantee that I'm relying on here. 

I also added a property test to exercise the parsing logic a bit more, with random amounts of newlines / carriage returns.

Reply:
Added back `is_empty()` check.